NebulaWorld/Combat/EnemyManager.cs

```diff
@@ -0,0 +1,170 @@
+﻿#region
+
+using System;
+using System.Collections.Generic;
+using NebulaModel.DataStructures;
+using NebulaModel.Packets.Combat.DFHive;
+using NebulaModel.Packets.Combat.GroundEnemy;
+#pragma warning disable IDE1006 // Naming Styles
+
+#endregion
+
+namespace NebulaWorld.Combat;
+
+public class EnemyManager : IDisposable
+{
+    public readonly ToggleSwitch IsIncomingRequest = new();
+
+    public readonly ToggleSwitch IsIncomingRelayRequest = new();
+
+    public readonly Dictionary<int, int[]> GroundTargets = new();
+
+    public const bool DISABLE_DFCommunicator = true;
+
+    private readonly Dictionary<int, DFGUpdateBaseStatusPacket> basePackets = [];
+    private readonly Dictionary<int, DFHiveUpdateStatusPacket> hivePackets = [];
+
+    public EnemyManager()
+    {
+        GroundTargets.Clear();
+        basePackets.Clear();
+        hivePackets.Clear();
+    }
+
+    public void Dispose()
+    {
+        GroundTargets.Clear();
+        basePackets.Clear();
+        hivePackets.Clear();
+        GC.SuppressFinalize(this);
+    }
+
+    public void GameTick(long gameTick)
+    {
+        if (!Multiplayer.Session.IsGameLoaded)
+        {
+            return;
+        }
+        // Place holder for future
+        var _ = gameTick;
+    }
+
+    public void BroadcastBaseStatusPackets(EnemyDFGroundSystem enemySystem, long gameTick)
+    {
+        var factoryIndex = enemySystem.factory.index;
+        var bases = enemySystem.bases;
+        for (var baseId = 1; baseId < bases.cursor; baseId++)
+        {
+            var dFbase = bases.buffer[baseId];
+            if (dFbase == null || dFbase.id != baseId) continue;
```

Comment:
Minor nitpick, I assume baseId can never be null, if this assumption is correct this can be simplified to:
```suggestion
            if (dFbase?.id != baseId) continue;
```